### PR TITLE
SignupProcessingScreen refinements (Fix to #6114)

### PR DESCRIPTION
**Pull request by coreh**
This PR fixes Issue #6114, by making it so that the `SignupFlowController` is reset before the user clicks on the Continue button.

It also adds two other improvements:

- The `SignupProcessingScreenI18n` stub is removed (We no longer need that, since the actual component landed)
- The `Sign Up` link in the masterbar is no longer displayed during signup. (So that user's can't navigate to `/signup` by accidentally clicking on it)

Test live: https://calypso.live/?branch=fix/issue-6114

### CODE REVIEW

File: client/layout/masterbar/logged-out.jsx [hunk 1]
```diff
@@ -13,7 +13,7 @@ import { localize } from 'i18n-calypso';
 import Item from './item';
 import config from 'config';
 
-const MasterbarLoggedOut = ( { title, translate } ) => (
+const MasterbarLoggedOut = ( { title, translate, showSignup = true } ) => (
```
Comment (michaeldcain) [inline client/layout/masterbar/logged-out.jsx]: Let's add `showSignup` to `.propTypes` in L44, and although I know they achieve the same thing, we should probably either define the default in `defaultProps` (L48 - like `title`), and remove it here, or define `title = ''` here, and remove `.defaultProps` for consistency.

**Comment by michaeldcain** [comment]
@coreh: Just a small edit, but otherwise is looks and works great. Thanks for the quick fix!

File: client/layout/masterbar/logged-out.jsx (hunk 1 above)
Comment (coreh) [inline client/layout/masterbar/logged-out.jsx]: Oh I missed that entirely. Adding it.

**Comment by coreh** [comment]
Fixed. I also moved `translate` to the end of the props object destructuring, just for organization's sake, since it seems to be added by the `localize` function, at the bottom.

**Comment by michaeldcain** [comment]
LGTM

**Pull request merged**